webpack.config.js

```diff
@@ -1,23 +1,57 @@
+/* eslint-disable global-require */
+const webpack = require('webpack');
 const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const CleanWebpackPlugin = require('clean-webpack-plugin');
+
 
 module.exports = {
   entry: ['./browser/src/react-redux/index.js'],
   mode: 'development',
   module: {
-    rules: [{
-      test: /js?$/,
-      exclude: /node_modules/,
-      use: {
-        loader: 'babel-loader',
+    rules: [
+      {
+        test: /jsx?$/,
+        exclude: /node_modules/,
+        use: {
+          loader: 'babel-loader',
+        },
+      },
+      {
+        test: /\.scss$/,
+        use: [
+          'style-loader',
+          'css-loader',
+          'sass-loader',
+        ],
       },
-    }],
+    ],
   },
   resolve: {
     extensions: ['.js'],
   },
   output: {
-    path: path.join(__dirname, 'browser/public/js'),
+    path: path.join(__dirname, 'browser/dist'),
     filename: 'scripts.js',
   },
   devtool: 'source-map',
+  devServer: {
+    contentBase: path.join(__dirname, 'browser/dist'),
+    historyApiFallback: true,
+    hot: true,
+    proxy: {
+      '/': 'http://localhost:3000',
+    },
+    port: 3001,
+  },
+  plugins: [
+    new webpack.HotModuleReplacementPlugin(),
+    new CleanWebpackPlugin([path.join(__dirname, 'browser/dist')]),
+    new HtmlWebpackPlugin({
+      inject: false,
```

Comment:
If we set this to `true`, it should automatically inject the styles and script bundles into index.html. Then we can set up `publicPath` based on whether we are in `prod` or `dev`. It _may_ set the public path automatically though.

I believe we can get rid of the `css` and `javascript` folders inside of `public` after this. Then, we might just want to rename `public` to `assets` to better describe what it's for. DM on Slack if you have any questions about this.

Reply:
`inject: false` is a required configuration for `html-webpack-plugin`. see https://github.com/jaketrent/html-webpack-template. The reason for this webpack plugin, is because I needed to add a `<div>` with `id=`app`` to the `index.html` file. Reason being, the React `index.js` file needs that div in order to render everything. `html-webpack-template` adds the the ability to include that needed div directly into the `index.html` file created by `html-webpack-plugin` at runtime.